### PR TITLE
updated inactivity timeout logic

### PR DESCRIPTION
**Pull request by DestinyofYeet**
example_options.ini: added space between text and #

After creating your pull request, tick these boxes if they are applicable to you.

- [X] I have tested my changes against the `review` branch (the latest developmental version), and this pull request is targeting that branch as a base
- [X] I have tested my changes on Python 3.8 or higher
- [X] I have ensured my code is formatted using [Black](https://github.com/psf/black)

----

### Description

Updated the inactivity timeout logic

### CODE REVIEW

File: musicbot/bot.py [hunk 1]
```diff
@@ -1393,6 +1393,26 @@ def _remove_url_from_autoplaylist(self, url):
         write_file(self.config.auto_playlist_file, self.autoplaylist)
         log.debug("Removed {} from autoplaylist".format(url))
 
+    async def handle_timeout(self, guild: discord.Guild):
+        event = self.server_specific_data[guild]["timeout_event"]
+
+        try:
+            log.debug(
```
Comment (BabyBoySnow) [inline musicbot/bot.py]: I'd prefer this to be log.info as it's not an bug issue or anything funny going on.

File: musicbot/bot.py [hunk 2]
```diff
@@ -1393,6 +1393,26 @@ def _remove_url_from_autoplaylist(self, url):
...
+                f"About to go to sleep for {self.config.leave_inactiveVCTimeOut} seconds"
+            )
+            await discord.utils.sane_wait_for(
+                [event.wait()], timeout=self.config.leave_inactiveVCTimeOut
+            )
+        except asyncio.TimeoutError:
+            log.debug("Timeout timed out, leaving channel")
```
Comment (BabyBoySnow) [inline musicbot/bot.py]: same thing here

File: musicbot/bot.py [hunk 3]
```diff
@@ -1393,6 +1393,26 @@ def _remove_url_from_autoplaylist(self, url):
...
+
+            await self.on_timeout_expired(guild.me.voice.channel)
+        else:
+            log.debug("Timeout event got set, stopping")
```
Comment (BabyBoySnow) [inline musicbot/bot.py]: same thing here

File: musicbot/bot.py [hunk 4]
```diff
@@ -1393,6 +1393,26 @@ def _remove_url_from_autoplaylist(self, url):
...
+        finally:
+            log.debug("Event cleared")
```
Comment (BabyBoySnow) [inline musicbot/bot.py]: same thing here

File: musicbot/bot.py [hunk 5]
```diff
@@ -4429,82 +4449,53 @@ async def gen_cmd_list(self, message, list_all_cmds=False):
...
-                    if str(before.channel.id) in str(self.config.autojoin_channels):
-                        log.info(
-                            f"Ignoring {before.channel.name} in {before.channel.guild} as it is a binded voice channel."
-                        )
-            elif after.channel:
-                if after.channel.id in timers:
-                    timers[after.channel.id].cancel()
-                    log.info(
-                        f"Cancelling timer for {after.channel.name} in {after.channel.guild} as channel is no longer inactive."
+                    log.debug("Channel is empty, should be handling disconnects now")
```
Comment (BabyBoySnow) [inline musicbot/bot.py]: log.info(f"{before.channel.name} has been detected as empty. Handling timeouts.")

File: musicbot/bot.py [hunk 6]
```diff
@@ -4429,82 +4449,53 @@ async def gen_cmd_list(self, message, list_all_cmds=False):
...
-                            lambda task: asyncio.ensure_future(
-                                self.on_timer_expired(before.channel)
-                            )
-                        )
-                        log.info(
-                            f"Started timer for inactive channel {before.channel.name} in {before.channel.guild}"
-                        )
-                    if str(before.channel.id) in str(self.config.autojoin_channels):
-                        log.info(
-                            f"Ignoring {before.channel.name} in {before.channel.guild} as it is a binded voice channel."
```
Comment (BabyBoySnow) [inline musicbot/bot.py:4489]: Removing this will make binded voice channels not work.

File: musicbot/bot.py [hunk 7]
```diff
@@ -4429,82 +4449,53 @@ async def gen_cmd_list(self, message, list_all_cmds=False):
...
-                if after.channel.id in timers:
-                    timers[after.channel.id].cancel()
-                    log.info(
-                        f"Cancelling timer for {after.channel.name} in {after.channel.guild} as channel is no longer inactive."
+                    log.debug("Channel is empty, should be handling disconnects now")
+                    await self.handle_timeout(guild)
+            elif after.channel and member != self.user:
+                if self.user in after.channel.members:
+                    log.debug(
+                        "Someone joined the channel again, should cancel the timer for disconnects"
```
Comment (BabyBoySnow) [inline musicbot/bot.py]: log.info(f"A user joined {after.channel.name}, cancelling timer."} 
I do like the rewording here though.

File: musicbot/bot.py [hunk 8]
```diff
@@ -114,7 +114,7 @@ def __init__(self, config_file=None, perms_file=None, aliases_file=None):
             "auto_paused": False,
             "availability_paused": False,
             "guild_id": None,
```
Comment (BabyBoySnow) [inline musicbot/bot.py]: We can probably remove this now.

**Review by BabyBoySnow — CHANGES_REQUESTED** [review]
Overall looks much better than before. Just needs some final touches imo.

File: musicbot/bot.py (hunk 8 above)
Comment (DestinyofYeet) [reply musicbot/bot.py]: Not removing `availability_paused` because it is still used in line 4668, 4674 and 4686

**Comment by DestinyofYeet** [comment]
Implemented changes :D

**Pull request merged**